### ISSUE TITLE
Find alternative solution to Javascript Interface Injection

### ISSUE THREAD

**Issue by psymoon** [mozilla-mobile/firefox-tv#1248]
https://github.com/mozilla-mobile/firefox-tv/blob/master/app/src/main/java/org/mozilla/focus/ext/EngineView.kt#L66

https://github.com/mozilla-mobile/firefox-tv/blob/master/app/src/main/java/org/mozilla/focus/ext/EngineView.kt#L73

Maybe something a-c can take care of?

**Comment by cpeterso** [comment]
@psymoon Does this MediaSession issue require new GeckoView APIs?

btw, EngineView's `addJavascriptInterface` and `removeJavascriptInterface` method definitions moved here:

https://github.com/mozilla-mobile/firefox-tv/blob/master/app/src/main/java/org/mozilla/tv/firefox/ext/EngineView.kt#L62-L75

**Comment by psymoon** [comment]
@pocmo @cpeterso yes that is correct, GeckoView doesn't have javascript injection API AFAIK

**Comment by pocmo** [comment]
@psymoon With us having more control over GeckoView: Would we prefer some kind of MediaSession API (instead of injecting JS and doing all that manually)?

**Comment by psymoon** [comment]
@pocmo yes, that would make more sense

**Comment by cpeterso** [comment]
I filed a bug requesting a GeckoView API, though we need more details about what the API would need to do and if there is a precedent WebView API we can copy.

https://bugzilla.mozilla.org/show_bug.cgi?id=1508371

**Comment by cpeterso** [comment]
In [GV bug 1508371](https://bugzilla.mozilla.org/show_bug.cgi?id=1508371#c6), @mcomella says that FFTV should be able to implement the MediaSession using GV's existing MediaDelegate and MediaElement APIs. I will close GV bug 1508371 and the FFTV team can track the MediaSession work in this issue.

**Comment by pocmo** [comment]
If the MediaDelegate is enough then this sounds like something we may want to implement in the/a component. Does this make sense @mcomella?

**Comment by cpeterso** [comment]
@athomasmoz: We'll need MediaSession support (either this FFTV issue or mozilla-mobile/android-components#1846) for FFTV+GV.

**Comment by liuche** [comment]
@pocmo it sounds like this is possible without the AC to expose this in `engineSession` bug https://github.com/mozilla-mobile/android-components/issues/1845 , but what's the timeline on exposing that in AC? Wondering if we should start working with using `MediaDelegate` or wait for AC.

**Comment by pocmo** [comment]
@liuche There's no ETA yet. I can try to get this on our radar in the planning meeting today.

> Wondering if we should start working with using MediaDelegate or wait for AC.

In this case I'd propose to not use the delegate directly and instead implement the AC API - No need to wait. :)

**Comment by psymoon** [comment]
@athomasmoz This is ready to be worked on by FFTV engineers (we would need to contribute to a-c) - https://github.com/orgs/mozilla-mobile/projects/26
